test/test_nn.py

```diff
@@ -15548,16 +15547,23 @@ def run_test(*shape):
                 xx = x.cpu().requires_grad_()
             yy = F.log_softmax(xx.float(), dim=-1).to(dtype)
             yy.backward(yy)
-            self.assertEqual(y, yy)
-            self.assertEqual(x.grad, xx.grad)
+            # workaround to reduce memory usage vs. self.assertEqual, see #84944
+            rtol = torch.testing._comparison._DTYPE_PRECISIONS[dtype][0]
+            atol = torch.testing._comparison._DTYPE_PRECISIONS[dtype][1]
+            atol = 0.001 if dtype is torch.half else atol
+            self.assertTrue(torch.allclose(y.cpu(), yy, rtol=rtol, atol=atol))
+            # x is half
+            self.assertTrue(torch.allclose(x.grad.cpu(), xx.grad,
+                                           rtol=torch.testing._comparison._DTYPE_PRECISIONS[torch.half][0],
+                                           atol=0.001))
```

Comment:
```suggestion
            rtol, atol = torch.testing._comparison.get_tolerances(torch.half, atol=1e-3)
            self.assertTrue(torch.allclose(x.grad.cpu(), xx.grad, rtol=rtol, atol=atol))
```

or if you really don't want to re-assign `rtol` and `atol`

```suggestion
            self.assertTrue(torch.allclose(x.grad.cpu(), xx.grad,
                                           **dict(zip(["rtol", "atol"], torch.testing._comparison.get_tolerances(torch.half, atol=1e-3)))))
```